girder/models/upload.py

```diff
@@ -291,22 +292,31 @@ def createUploadToFile(self, file, user, size, reference=None,
             unspecified, the current assetstore is used.
         """
         from girder.utility import assetstore_utilities
+        from .item import Item
 
         assetstore = self.getTargetAssetstore('file', file, assetstore)
         adapter = assetstore_utilities.getAssetstoreAdapter(assetstore)
         now = datetime.datetime.utcnow()
 
-        upload = {
-            'created': now,
-            'updated': now,
-            'userId': user['_id'],
-            'fileId': file['_id'],
-            'assetstoreId': assetstore['_id'],
-            'size': size,
-            'name': file['name'],
-            'mimeType': file['mimeType'],
-            'received': 0
-        }
+        if 'attachedToId' in file:
+            parent = Item().findOne({'_id': file['attachedToId']})
```

Comment:
The `attachedToType` isn't necessarily an item, so this should probably be `parent = ModelImporter.model(file['attachedToType']).findOne({'_id': file['attachedToId']})`